### DISCUSSION TITLE
Checking multiple Flipper-flags at once

### DISCUSSION THREAD

**Discussion by davidwessman** [jnunemaker/flipper#724] [Q&A] [answered]
Hello!

We have a use-case where we need to serialize the Flipper-state for one record for a limited number of flags.
We currently call `Flipper.enabled?(name, record)` in a loop.

Would it be possible to have the ability to pass multiple keys at once? `Flipper.enabled?(%i[a, b, c], record)`
Or some way to get all active flags for one record?

We are using the ActiveRecord-adapter.

**Comment by bkeepers** [comment]
Hi @davidwessman 

> We have a use-case where we need to serialize the Flipper-state for one record for a limited number of flags.
> We currently call `Flipper.enabled?(name, record)` in a loop.

bCan you explain more about what the concern is with calling Flipper in a loop? If you are using Rails and the default setup, all calls to Flipper are preloaded and memoized, so there is no significant overhead for repeated calls and only one database query.

> Would it be possible to have the ability to pass multiple keys at once? `Flipper.enabled?(%i[a, b, c], record)`
> Or some way to get all active flags for one record?

You would just have to collect these manually using `%i[a b c]).map { |feature| Flipper.enabled?(feature, record) }`. Again, if you're using the default configuration with Rails, this should only result in a single query.

Here's more info on memoization: https://www.flippercloud.io/docs/optimization

**Reply by davidwessman** [reply]
Thank you for a quick answer!

So the first call to `Flipper.enabled?(:feature_a, user_a)` will do one query for the feature and memoize the results for the same **feature** for all actors? E.g. `Flipper.enabled?(:feature_b, user_b)` will not use the memoized value?

This is not our main performance problem in any way, so I think we will keep track of removing our flipper-flags regularly 🙂

**Reply by bkeepers** [reply, marked as answer]
Flipper is configured to preload and memorize all feature data before each request.  So whether you check zero feature flags, or 1000 with multiple actors, it only makes one query to the database. The preloading will likely need tuned on large apps with a lot of features and actors to only selectively load features that are commonly used.
